### PR TITLE
Adding support for user validation

### PR DESCRIPTION
**Pull request by apahim**
When the `GITHUB_USERS` environment variable is present, the user
validation will be enabled. The `GITHUB_USERS` has to carry a list of
colon-separated user names that will have their requests served.

### CODE REVIEW

**Comment by maorfr** [comment]
Ah!

Should there be any changes to the openshift template?

File: ghmirror/decorators/checks.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+        user_info = resp.json()
+        # At this point we have the authorized_users list and the
+        # user information from Github. If there's a match, we just
+        # return the decorated function
+        if user_info['login'] in authorized_users:
+            return function(*args, **kwargs)
+
+        # No match means user is not authorized
+        return flask.Response('User is an authorized user', 401)
```
Comment (rporres) [inline ghmirror/decorators/checks.py]: nit: The message may be misleading. Shouldn't it say something like User not being authorized as the comment suggests :)

File: README.md [hunk 2]
```diff
@@ -103,3 +103,17 @@ Run the development server:
 ```
 $  make run
 ```
+
+## User Validation
+
+To enable the user validation, the `GITHUB_USERS` environment variable
+should be available to the server. The `GITHUB_USERS` is a colon-separated
+list of authorized users to have their requests to the Guthub Mirror served.
```
Comment (rporres) [inline README.md]:
```suggestion
list of authorized users to have their requests to the Github Mirror served.
```

File: ghmirror/decorators/checks.py (hunk 1 above)
Comment (apahim) [reply ghmirror/decorators/checks.py]: Fixed!

File: README.md (hunk 2 above)
Comment (apahim) [reply README.md]: Fixed!

**Comment by apahim** [comment]
Thanks for the review guys. Now waiting for https://github.com/app-sre/qontract-reconcile/pull/530 to go through.

**Pull request merged**